tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -137,15 +139,25 @@ def _validate(self):
                 if image not in self.images:
                     raise Exception(f"Invalid image reference in test suite {suite.name}: Can't find {image} in images.yml")
 
-            # If the suite specifies a location, validate that the images it uses are available in that location
-            if suite.location != '':
-                for suite_image in suite.images:
-                    for image in self.images[suite_image]:
-                        # If the image has a location restriction, validate that it is available on the location the suite must run on
-                        if image.locations:
-                            locations = image.locations.get(self.__cloud)
-                            if locations is not None and not any(suite.location in l for l in locations):
-                                raise Exception(f"Test suite {suite.name} must be executed in {suite.location}, but <{image.urn}> is not available in that location")
+            # If the suite specifies a cloud specific location, validate that the images it uses are available in that location
+            for suite_location in suite.locations:
+                if self.__cloud in suite_location:
```

Comment:
since the intention is to skip the cloud, I think the code would be clearer doing something like

```
if suite location does not start with "cloud:":
   continue
```

instead of setting to empty string (which is confusing) and then checking for empty string